### ISSUE TITLE
Dynamic building / loading of module

### ISSUE THREAD

**Issue by anitakrueger** [chobits/ngx_http_proxy_connect_module#109]
### Ⅰ. Issue Description

I've tried to build the module using `--add-dynamic-module`, but am getting the following error when trying to load it:
```
[emerg] dlopen() "/usr/lib64/nginx/modules/ngx_http_proxy_connect_module.o" failed (/usr/lib64/nginx/modules/ngx_http_proxy_connect_module.o: only ET_DYN and ET_EXEC can be loaded) in /etc/nginx/nginx.conf:11
```

### Ⅱ. Describe what happened

The module can't be loaded dynamically.

### Ⅲ. Describe what you expected to happen

Loading the module dynamically, so I don't have to build nginx on the box I need the module to be installed.

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. Configure the nginx sources using `--add-dynamic-module=../ngx_http_proxy_connect_module`
2. Find the compiled ngx_http_proxy_connect_module.o and try to use it with the same nginx version (same arguments).
3. Load it in nginx.conf like this:
```
load_module /usr/lib64/nginx/modules/ngx_http_proxy_connect_module.o;
```

### Ⅴ. Anything else we need to know?

1. If applicable, add nginx  [debug log](http://nginx.org/en/docs/debugging_log.html).
2. I'm using nginx version 1.15.9 and used the appropriate patch
3. Seems like this was opened before but then closed: https://github.com/chobits/ngx_http_proxy_connect_module/issues/7

### Ⅵ. Environment:

1. Tengine/Nginx/OpenResty version (use `sbin/nginx -v`):
```
nginx version: nginx/1.15.9
```
2. Which [patch](https://github.com/chobits/ngx_http_proxy_connect_module#select-patch) do you use?
```
proxy_connect_rewrite_101504.patch
```

**Comment by chobits** [comment]
Thanks for your report, dynamic loading is not supported currently.

Will support this feature soon.

**Comment by anitakrueger** [comment]
Thank you @chobits 
I did create PR https://github.com/chobits/ngx_http_proxy_connect_module/pull/110 which will make the module build dynamically and create a shared object file. It does load in nginx and there are no errors there. Unfortunately I am having trouble getting it to proxy correctly, but I don't know if that is because of my configuration or the module. I was hoping you would have some automated test cases that could test this a bit better.

**Comment by chobits** [comment]
Thanks for your contribution, I have merged your PR #110 .
